pkg/kudoctl/packages/reader/parser.go

```diff
@@ -10,10 +11,13 @@ import (
 )
 
 const (
-	operatorFileName      = "operator.yaml"
-	templateFileNameRegex = "templates/.*.yaml"
-	paramsFileName        = "params.yaml"
-	APIVersion            = "kudo.dev/v1beta1"
+	operatorFileName = "operator.yaml"
+	paramsFileName   = "params.yaml"
+	APIVersion       = "kudo.dev/v1beta1"
+)
+
+var (
+	templateFileNameRegex = filepath.Join("templates", ".*.yaml")
```

Comment:
The name indicates that this is a regex... Is there any issue here with having a `\` instead of a `/`? And don't we need to escape the `.` before "yaml"?  Unescaped, it would match something like "templates/weirdnameyaml", or is there something special in Go regexps?

Reply:
👍 to all of that.
`parsePackageFile` could be refactored to use [filepath.Walk](https://golang.org/pkg/path/filepath/#Walk) to avoid using a regex here.

Reply:
@nfnt `parsePackageFile` *is already* called from a file walker. The problem, here: you get a filepath like `/go/src/github.com/kudobuilder/operators/repository/cowsay/operator/templates/container.yaml` or `./templates/container.yaml` and you want to make sure it's a "kudo operator template file"

Reply:
Thanks for clarifying, seeing this now in `read_dir.go`, was looking at `parser.go` in isolation.

Reply:
With or without escaping, windows paths wouldn't match (I tested it on windows). So, here is another take on it.